Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [17]:
import numpy as np
from math import factorial as f


In [18]:
x = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

среднее арифметическое

In [19]:
x.size


20

In [20]:
x.sum()

1306

In [21]:
arithmetic_mean = x.sum()/x.size

In [22]:
arithmetic_mean

65.3

среднее квадратичное отклонение

In [23]:
mean_square_deviation = (np.sum((x - arithmetic_mean)**2) / x.size)**0.5

In [24]:
mean_square_deviation

30.823854398825596

смещенную и несмещенную оценки дисперсий для данной выборки

смещенная

In [25]:
np.sum((x - arithmetic_mean)**2) / x.size

950.11

не смещенная

In [26]:
np.sum((x - arithmetic_mean)**2) / (x.size-1)

1000.1157894736842

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [84]:
def combinations(n,k):
    return f(n) / (f(k) * f(n-k))

у нас 3 варианта событий: 0Б / 3Б+1ч, 1Б/2Б+2ч, 2Б/1Б+3ч

событие 1 0Б / 3Б+1ч

In [92]:
x1= (combinations (3,2)* combinations (5,0))/combinations (8,2) * (combinations (5,3)* combinations (7,0))/ combinations (12,4)

In [93]:
x1

0.0021645021645021645

событие 2 1Б/2Б+2ч

In [94]:
x2=(combinations (3,1)* combinations (5,1))/combinations (8,2) * (combinations (5,2)* combinations (7,2))/ combinations (12,4)

In [96]:
x2

0.22727272727272727

событие 3 2Б/1Б+3ч

In [97]:
x3=(combinations (3,0)* combinations (5,2))/combinations (8,2) * (combinations (5,1)* combinations (7,3))/ combinations (12,4)

In [98]:
x3

0.12626262626262627

In [99]:
x=x1+x2+x3

In [100]:
x

0.35569985569985574

Вероятность что все 3 белые 0.35569985569985574


На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [101]:
x = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6


In [102]:
x1 = (1/3 * 0.9) / x

In [103]:
x1

0.391304347826087

Первый спортсмен х1 = 0,3913

In [104]:
x2 = (1/3 * 0.8) / x

In [105]:
x2

0.3478260869565218

Второй спортсмен х2 = 0.3478260869565218

In [107]:
x3 = (1/3 * 0.6) / x

In [108]:
x3

0.2608695652173913

Третий спортсмен х3 = 0.2608695652173913

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Всего студентов 100 процентов = Фак А 25 процентов, факультет Б 25 процентов и факультет С - 50 процентов
Задача решается аналогичным способом

In [148]:
x= 1/4 * 0.8 + 1/4 * 0.7 + 1/2 * 0.9

In [149]:
x

0.825

In [115]:
FA = (1/4*0.8) / x 

In [116]:
FA

0.24242424242424246

Фак. А - вероятность 0,24

In [117]:
FB = (1/4*0.7) / x 

In [127]:
FB

0.21212121212121213

Фак. В - вероятность 0,24

In [138]:
(1/2*0.9) / x 

0.5454545454545455

Фак. C - вероятность 0,54

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [154]:
p1*p2+p1*p3+p2*p3-3*p1*p2*p3

In [208]:
P =  (1*0.1)*(1*0.2)+(1*0.1)*(1*0.25)+ (1*0.2)*(1*0.25) - ( 3*(1*0.1)*(1*0.2)*(1*0.25) )

In [209]:
P

0.08

P = Вероятность что только две детали выйдут из строя ровна 0,8

In [201]:
P1 = (1*0.1)*(1*0.2)*(1*0.25)

In [202]:
P1

0.005000000000000001

P = Вероятность что детали выйдут из строя ровна 0,005

In [203]:
P2 = (1*0.1)+(1*0.2)+(1*0.25)

In [204]:
P2

0.55

Вероятность что выйдет их строя хотя бы одна деталь 0.55

In [212]:
P3 = P2*P

In [213]:
P3

0.044000000000000004

In [ ]:
Вероятность что выйдет от 1 до 2х деталей ровна 0,44